rotkehlchen/db/filtering.py

```diff
@@ -1143,6 +1144,25 @@ def make(
                 operator='IN',
             ))
 
+        if location_labels_or_validator_indices:
```

Comment:
Why not to split it into 2 filters? One for location labels and the other one for validator indices

Reply:
The reason to have this extra filter is because as for how filters work right now if you provide at the same time `location_labels` and `validator_indices` the filter will do `... AND location_label IN location_labels AND validator_index IN validator_indices AND ...`

What you really want is by filtering by some validator index and some address is `... AND (location_label IN location_labels OR validator_index IN validator_indices) AND...`. Notice the difference from `AND` to `OR`. For this we need to use a nested filter and this is why I have this extra logic here

Reply:
Yeah, I get it. My suggestion, as said here https://github.com/rotki/rotki/pull/5950#discussion_r1182266551, is to handle this condition in this file, in `filtering.py`